lib/view_component/base.rb

```diff
@@ -319,6 +331,15 @@ def __vc_render_in_block_provided?
       defined?(@view_context) && @view_context && @__vc_render_in_block
     end
 
+    def __vc_render_template(rendered_template)
+      # Avoid allocating new string when output_preamble and output_postamble are blank
+      if output_preamble.blank? && output_postamble.blank?
```

Comment:
Are there any chances only one of them would be blank?

Reply:
yes they can be individually toggled

Reply:
Not sure, if I'm contributing to this. If it's safe to have one of them blank (since we don't have a statement to cover this having only one of them blank), then isn't it also safe to do the else statement `safe_output_preamble + rendered_template + safe_output_postamble` if both of them are blank?

Reply:
don't worry to much about the established code, a look at the cache mechanism is where the bulk of the eyes are needed